src/blk/kernel/KernelDevice.cc

```diff
@@ -780,6 +781,7 @@ void KernelDevice::_discard_thread(uint64_t tid)
       if (thr->stop && !discard_threads.empty())
         break;
 
+      std::this_thread::sleep_for(std::chrono::milliseconds(cct->_conf->bdev_debug_discard_sleep));
```

Comment:
Is sleeping for 0 is an equivalent of not sleeping at all?
Wouldn't it be like rounded up.
Also, if it makes a syscall with 0 would be bad.
Solution: wrap it with `if(cct->.. != 0)`